HealthSDK/GiniHealthSDK/Sources/GiniHealthSDK/Core/PaymentComponent/PaymentComponentsController.swift

```diff
@@ -8,28 +8,52 @@
 import UIKit
 import GiniHealthAPILibrary
 
+/**
+ Data structure protocol for the stored invoices/documents needed for Payment Component Controller
+ */
 public protocol GiniDocument {
+    /// The document's unique identifier.
     var documentID: String { get set }
+    /// The document's amount to pay.
     var amountToPay: String? { get set }
+    /// The document's payment due date.
     var paymentDueDate: String? { get set }
+    /// The document's recipient name.
     var recipient: String? { get set }
+    /// Boolean value that indicates if the document is payable. This is obtained by calling the checkIfDocumentIsPayable method.
     var isPayable: Bool? { get set }
+    /// Store payment provider for each document/invoice. Payment provider is obtained by PaymentComponentsController and it's only stored here if the payment provider is installed
     var paymentProvider: PaymentProvider? { get set }
 }
 
+/**
+ Delegate to inform about the current status of the Payment Components Controller.
+ Makes use of callback for handling payment providers request.
+ */
 public protocol PaymentComponentsControllerProtocol: AnyObject {
     func isLoadingStateChanged(isLoading: Bool) // Because we can't use Combine, iOS 11 supported
     func didFetchedPaymentProviders(_ paymentProviders: PaymentProviders)
     func didReceivedErrorOnPaymentProviders(_ error: GiniHealthError)
 }
 
+/**
+ The `PaymentComponentsController` class allows control over the payment components.
+ */
 public final class PaymentComponentsController: NSObject {
+    /// handling the Payment Component Controller delegate
     public weak var delegate: PaymentComponentsControllerProtocol?
+    /// handling the Payment Component view delegate
     public weak var viewDelegate: PaymentComponentViewProtocol?
 
     private var giniHealth: GiniHealth
     private var paymentProviders: PaymentProviders = []
 
+    /**
+     Returns a Payment Component Controller instance
```

Comment:
[suggestion]: I will rewrite it like this:
```
This function returns an instance of the Payment Component Controller.

    Parameters:
        giniHealth: An instance of GiniHealth initialized with GiniHealthAPI.
    note: Instantiating this class also triggers the loading of available payment providers, which are stored internally for future use.
```

Reply:
@ValentinaIancu-Gini  we use in our inline documentation "Returns ..."  😉